### PR TITLE
fix(container-loader): Return codeDetails.name

### PR DESCRIPTION
**Pull request by sonalideshpandemsft**
This PR fixes a minor bug found in this PR: https://github.com/microsoft/FluidFramework/pull/21542/files#r1648285464

### CODE REVIEW

**Review by ChumpChief — APPROVED** [review]

File: packages/loader/container-loader/src/contracts.ts [hunk 1]
```diff
@@ -217,7 +217,7 @@ export const getPackageName = (
 	// eslint-disable-next-line @typescript-eslint/no-explicit-any
 	let containerPackageName: any;
 	if (codeDetails && "name" in codeDetails) {
-		containerPackageName = codeDetails;
+		containerPackageName = codeDetails.name;
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/contracts.ts:220]: can we add a test to avoid similar regressions in the future?

Comment (anthony-murphy) [reply packages/loader/container-loader/src/contracts.ts:220]: a unit test of just this function should be trivial to add.

Comment (sonalideshpandemsft) [reply packages/loader/container-loader/src/contracts.ts:220]: i'll add unit tests

Comment (sonalideshpandemsft) [reply packages/loader/container-loader/src/contracts.ts]: while adding the unit tests, I noticed the `codeDetails` types should include `IContainerPackageTypes`. Without this typing, the first `if` check would lose its meaning.

```ts
export const getPackageName = (
 codeDetails: IFluidCodeDetails | IContainerPackageInfo | undefined,
): IContainerPackageInfo => {
 // TODO: use a real type
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 let containerPackageName: any;
 if (codeDetails && "name" in codeDetails) {
  containerPackageName = codeDetails.name;
 } else if (isFluidPackage(codeDetails?.package)) {
  containerPackageName = codeDetails?.package.name;
 } else {
  containerPackageName = codeDetails?.package;
 }
 // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
 return { name: containerPackageName };
};
```

`IContainerPackageName` is not passed today which probably means the first check should be removed. 

please correct me if i'm missing anything here.

Comment (ChumpChief) [reply packages/loader/container-loader/src/contracts.ts]: Removal sounds correct to me - if this has been "broken" for over 2 years, I think that's proof that it's unused and safe to remove.

**Pull request merged**